dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataAnalysisController.java

```diff
@@ -171,6 +175,17 @@ public class DataAnalysisController {
       throw new WebMessageException(badRequest("No organisation unit defined"));
     }
 
+    int maxResults =
+        validationRulesAnalysisParams.getMaxResults() != null
+            ? validationRulesAnalysisParams.getMaxResults()
+            : ValidationService.MAX_INTERACTIVE_ALERTS;
+
+    final int MAX_ALLOWED_RESULTS = settingsProvider.getCurrentSettings().getDataQualityMaxLimit();
+
+    if (maxResults <= 0 || maxResults > MAX_ALLOWED_RESULTS) {
+      throw new BadRequestException("maxResults must be between 1 and " + MAX_ALLOWED_RESULTS);
```

Comment:
Use "Max results" instead of "maxResults" in error message which is what is typically used.